fsmonitor.c

```diff
@@ -25,6 +26,18 @@ static void fsmonitor_ewah_callback(size_t pos, void *is)
 	ce->ce_flags &= ~CE_FSMONITOR_VALID;
 }
 
+static int fsmonitor_hook_version()
+{
+	int hook_version = -1;
+	if (!git_config_get_int("core.fsmonitorHookVersion", &hook_version)) {
+		if (hook_version != HOOK_INTERFACE_VERSION1 &&
+		    hook_version != HOOK_INTERFACE_VERSION2)
+			hook_version = -1;
```

Comment:
We could write this as:

```c
int hook_version;

if (git_config_get_int("core.fsmonitorhookversion", &hook_version) ||
    (hook_version != HOOK_INTERFACE_VERSION1 &&
     hook_version != HOOK_INTERFACE_VERSION2))
        return -1;
return hook_version;
```

What do you think?

Reply:
Personally, I think I'd go for something like:

    int hook_version;
    
    if (git_config_get_int(..., &hook_version))
            return -1;
    if (hook_version == V1 ||
        hook_version == V2)
            return hook_version;
    warning(....); // maybe / maybe not
    return -1;

But that is just another opinion.

Johannes' version avoids the compile-time initialization, but also makes it fuzzy as to whether hook_version has been written to at the final return.   In both Kevin's and Johannes' version, the negation of the V1 and/or V2 clauses makes for a little more difficult reading.  IMHO.